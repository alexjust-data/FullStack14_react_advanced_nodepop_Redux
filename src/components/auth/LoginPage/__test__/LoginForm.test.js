import React from 'react';
import { render } from '@testing-library/react';
import LoginForm from '../LoginForm';

describe('LoginForm', () => {
  it('should match snapshot', () => {
    //proveer una función "mock" (un sustituto o simulación) para el prop onSubmit 
    //usando jest.fn(). Esta es una forma de crear una función que no tiene una 
    //implementación específica, pero verificar se llamé, con qué argumentos, etc. 
    const mockSubmit = jest.fn();
    const { asFragment } = render(
      <LoginForm onSubmit={mockSubmit} isLoading={false} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
