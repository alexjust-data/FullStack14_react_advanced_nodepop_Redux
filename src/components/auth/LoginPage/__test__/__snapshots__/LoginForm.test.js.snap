// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should match snapshot 1`] = `
<DocumentFragment>
  <form>
    <input
      name="email"
      value=""
    />
    <input
      name="password"
      type="password"
      value=""
    />
    <input
      name="remember"
      type="checkbox"
    />
    <button
      disabled=""
    >
      Login
    </button>
  </form>
</DocumentFragment>
`;
